src/porepy/models/initial_condition.py

```diff
@@ -0,0 +1,117 @@
+"""Module containing mixins for providing initial values during a model setup."""
+
+from __future__ import annotations
+
+import porepy as pp
+
+
+class InitialConditionMixin(pp.PorePyModel):
```

Comment:
@IvarStefansson Can you have a look at this concept, and its use in existing models (search for inheritance from this class). A bit of background: Compositional flow introduces secondary variables to a much larger extent than what we have had before, and this implies more ellaborate initialization procedures, hence the need to work more in a more structured manner with this. There is a hint of this in the example starting 10 lines below from here.